core/src/main/java/org/elasticsearch/search/aggregations/metrics/percentiles/PercentilesParser.java

```diff
@@ -36,20 +42,25 @@ public PercentilesParser() {
 
     @Override
     public String type() {
-        return InternalPercentiles.TYPE.name();
+        return InternalTDigestPercentiles.TYPE.name();
     }
 
     @Override
-    protected String keysFieldName() {
-        return "percents";
+    protected ParseField keysField() {
+        return PERCENTS_FIELD;
     }
-    
+
     @Override
-    protected AggregatorFactory buildFactory(SearchContext context, String aggregationName, ValuesSourceConfig<Numeric> valuesSourceConfig, double[] keys, double compression, boolean keyed) {
+    protected AggregatorFactory buildFactory(SearchContext context, String aggregationName, ValuesSourceConfig<Numeric> valuesSourceConfig,
+            double[] keys, PercentilesMethod method, Double compression, Integer numberOfSignificantValueDigits, boolean keyed) {
         if (keys == null) {
             keys = DEFAULT_PERCENTS;
         }
-        return new PercentilesAggregator.Factory(aggregationName, valuesSourceConfig, keys, compression, keyed);
+        if (method == PercentilesMethod.TDIGEST) {
+            return new TDigestPercentilesAggregator.Factory(aggregationName, valuesSourceConfig, keys, compression, keyed);
+        } else {
+            return new HDRPercentilesAggregator.Factory(aggregationName, valuesSourceConfig, keys, numberOfSignificantValueDigits, keyed);
+        }
```

Comment:
it can't happen in practice, but I would like better something that would do if (tdigest) {} else if (hdr) {} else { throw new AssertionError(); }